### PR TITLE
Add SuperResolution example and test

### PR DESCRIPTION
**Pull request by bbernhar**
Adds SuperResolution to WebNN!

### CODE REVIEW

**Comment by bbernhar** [comment]
@fujunwei PTAL!

File: examples/SuperResolution/SuperResolution.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        fusedOptions.activation = builder.ReluOperator();
+    }
+
+    return builder.Conv2d(input, convWeights, fusedOptions.AsPtr());
+}
+
+const wnn::Operand SuperResolution::LoadNchw(const wnn::GraphBuilder& builder, bool softmax) {
+    const wnn::Operand input = utils::BuildInput(builder, "input", {1, 1, 224, 224});
+
+    // TODO: Does kernel_shape (5,5) need to be specified?
```
Comment (fujunwei) [inline examples/SuperResolution/SuperResolution.cpp:76]: The shape is got from npy files seeing L.26.

File: examples/SuperResolution/SuperResolution.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    std::vector<int32_t> permutation = {0, 1, 4, 2, 5, 3};
+    transpose1Options.permutation = permutation.data();
+    transpose1Options.permutationCount = permutation.size();
+    const wnn::Operand transpose1 = builder.Transpose(reshape1, &transpose1Options);
+
+    const std::vector<int32_t> newShape2 = {-1, 1, 672, 672};
+    const wnn::Operand reshape2 = builder.Reshape(transpose1, newShape2.data(), newShape2.size());
+
+    const wnn::Operand output = reshape2;
+    return output;
```
Comment (fujunwei) [inline examples/SuperResolution/SuperResolution.cpp:121]: Merge L.118 and L.121 to `return builder.Reshape(transpose1, newShape2.data(), newShape2.size()); ` that is output operand.

File: examples/SuperResolution/SuperResolution.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 The WebNN-native Authors
```
Comment (fujunwei) [inline examples/SuperResolution/SuperResolution.h:1]: Ditto.

File: src/webnn/native/dmlx/GraphDMLX.cpp [hunk 4]
```diff
@@ -1270,7 +1270,7 @@ namespace webnn::native::dmlx {
         DAWN_ASSERT(mExpression.find(inputOperand) != mExpression.end());
         ::dml::Expression input = mExpression.at(inputOperand);
         auto newShape = reshape->GetNewShape();
-        if (newShape.size() > DML_TENSOR_DIMENSION_COUNT_MAX) {
+        if (newShape.size() > DML_TENSOR_DIMENSION_COUNT_MAX1) {
```
Comment (fujunwei) [inline src/webnn/native/dmlx/GraphDMLX.cpp:1273]: @mingmingtasd Please update it to dml backend.

File: src/webnn/tests/end2end/models/SuperResolutionNchw.cpp [hunk 5]
```diff
@@ -0,0 +1,43 @@
+// Copyright 2021 The WebNN-native Authors
```
Comment (fujunwei) [inline src/webnn/tests/end2end/models/SuperResolutionNchw.cpp:1]: Ditto.

**Review by fujunwei — APPROVED** [review]
Thanks. LGTM and left some comments.

Could you please add a ReadMe for the example like [1]?

[1] https://github.com/webmachinelearning/webnn-native/blob/main/examples/MobileNetV2/README.md

File: src/webnn/native/dmlx/GraphDMLX.cpp (hunk 4 above)
Comment (mingmingtasd) [reply src/webnn/native/dmlx/GraphDMLX.cpp:1273]: OK, got it, will do it when re-writing DML backend. @fujunwei @miaobin

File: examples/SuperResolution/SuperResolution.cpp [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+// limitations under the License.
+
+#include "examples/SuperResolution/SuperResolution.h"
+
+#include "common/Log.h"
+
+SuperResolution::SuperResolution() : ExampleBase() {
+}
+
+const wnn::Operand SuperResolution::BuildConstantFromNpy(const wnn::GraphBuilder& builder,
```
[the remark this answers is not in the document]
Comment (mingmingtasd) [reply examples/SuperResolution/SuperResolution.cpp:22]: Thanks! I will update after this PR merged. @fujunwei

File: examples/SuperResolution/SuperResolution.h (hunk 3 above)
Comment (bbernhar) [reply examples/SuperResolution/SuperResolution.h:1]: Done.

File: examples/SuperResolution/SuperResolution.cpp [hunk 7]
```diff
@@ -0,0 +1,122 @@
+// Copyright 2021 The WebNN-native Authors
```
[the remark this answers is not in the document]
Comment (bbernhar) [reply examples/SuperResolution/SuperResolution.cpp:1]: Done.

File: src/webnn/native/dmlx/GraphDMLX.cpp (hunk 4 above)
Comment (bbernhar) [reply src/webnn/native/dmlx/GraphDMLX.cpp:1273]: Ack.

File: src/webnn/tests/end2end/models/SuperResolutionNchw.cpp (hunk 5 above)
Comment (bbernhar) [reply src/webnn/tests/end2end/models/SuperResolutionNchw.cpp:1]: Done.

File: examples/SuperResolution/SuperResolution.cpp [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "examples/SuperResolution/SuperResolution.h"
+
+#include "common/Log.h"
```
[the remark this answers is not in the document]
Comment (bbernhar) [reply examples/SuperResolution/SuperResolution.cpp:17]: Done.

File: examples/SuperResolution/SuperResolution.cpp (hunk 2 above)
Comment (bbernhar) [reply examples/SuperResolution/SuperResolution.cpp:121]: Done.

File: examples/SuperResolution/SuperResolution.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (bbernhar) [reply examples/SuperResolution/SuperResolution.cpp:22]: Ack.

File: examples/SuperResolution/SuperResolution.cpp (hunk 1 above)
Comment (bbernhar) [reply examples/SuperResolution/SuperResolution.cpp:76]: Ack

**Comment by bbernhar** [comment]
@fujunwei I don't have the example building yet (for the README), only the test. Will come next if that's OK.

**Comment by fujunwei** [comment]
make sense, merge it, thanks.

**Pull request merged**